nixos/lib/make-disk-image.nix

```diff
@@ -45,7 +49,9 @@ let
     raw   = "img";
   };
 
-  nixpkgs = lib.cleanSource pkgs.path;
+  compress = optionalString compressed (assert format == "qcow2"; "-c");
```

Comment:
Rather than putting an assert here, maybe we could just make `format` be one of `vpc`, `qcow2`, `qcow2-compressed`, and `raw`? From a types perspective it feels cleaner to have a 4-element enum than a boolean paired with a 3-element enum where 2 of the choices are invalid.

Reply:
This makes the naming difficult: we shouldn't rename `format` parameter as it is a part of the API, and there is no better name for the actual format as understood by `qemu-img`. Is this better: https://github.com/NixOS/nixpkgs/pull/31535/commits/ab53443ef0d636bd7d549e93e8792a52d88b92b5?

Reply:
An alternative solution is to rename `compressed` into `qcow2-compressed`:
```diff
-  compress = optionalString compressed (assert format == "qcow2"; "-c");
+  compress = optionalString (format == "qcow2" && qcow2-compressed) "-c";
```
(This facilitates making qcow2 be compressed by default, although I'm not sure if that is useful.)

Reply:
I quite like the `qcow2-compressed` version myself. Thanks!